cnxepub/models.py

```diff
@@ -82,36 +81,22 @@ def utf8(item):
         return item
 
 
-def _sanitize_xml(raw_xml, recover=True):
-    """Wraps the XML and sanitizes the namespaces."""
-    xml_parser = etree.XMLParser(ns_clean=True, recover=recover)
-    xml = etree.fromstring(XML_WRAPPER.format(utf8(raw_xml)), xml_parser)
-    result = utf8(etree.tostring(xml))
-    result = result[result.find('>')+1:result.rfind('<')].strip()
-    return result
-
-
 def content_to_etree(content):
+    if not content:
+        return etree.XML('<div/>')
+    xml_parser = etree.XMLParser(ns_clean=True)
+    tree = etree.XML(content, xml_parser)
     try:
-        xml_parser = etree.XMLParser(ns_clean=True, recover=True)
-        tree = etree.HTML(content, xml_parser)
         bod = tree.xpath('//*[self::body|self::x:body]',
                          namespaces={'x': 'http://www.w3.org/1999/xhtml'})[0]
-        bod.tag = 'div'
+        bod.tag = '{http://www.w3.org/1999/xhtml}div'
         return bod
-    # Reeeally broken HTML, or no body
-    except (etree.Error, AttributeError, IndexError):
-        content = _sanitize_xml(content)
-        return lxml.html.fragment_fromstring(content, 'div')
+    except IndexError:
```

Comment:
Since this is a generic/base exception, it is not 100% clear to me what could be causing it. So I'd suggest adding a comment explaining what might cause it exactly.

Reply:
I think this is meant to catch the `[0]` in:

```
        bod = tree.xpath('//*[self::body|self::x:body]',
￼	                         namespaces={'x': 'http://www.w3.org/1999/xhtml'})[0]
```

Reply:
Might be better to do `if bod:` instead of try except `IndexError`?

Reply:
Right, notice that the diff in that case is not adding the catching of the `IndexError`, it's reducing the number of classes of errors caught (from `etree.Error`, `AttributeError`, `IndexError` to just IndexError) I agree it's not the best way to write this code, I can recast it as Karen suggests.